src/blueapi/cli/cli.py

```diff
@@ -21,54 +19,36 @@ def main(ctx) -> None:
     if ctx.invoked_subcommand is None:
         print("Please invoke subcommand!")
 
+    config_loader = ConfigLoader(ApplicationConfig)
+
+    ctx.ensure_object(dict)
+    ctx.obj["config"] = (
+        config_loader.load_from_yaml(DEFAULT_YAML_PATH)
+        if DEFAULT_YAML_PATH.exists()
+        else config_loader.load()
```

Comment:
is config_load.load() with empty dict a valid configuration?

Reply:
yep; the schema given into `ConfigLoader` must be of type `BaseSettings`. These can have defaults or default factories. In the case of the ApplicationContext, every field has a default or default factory.